synapse/handlers/federation_event.py

```diff
@@ -356,6 +356,11 @@ async def on_send_membership_event(
                 )
 
         # all looks good, we can persist the event.
+
+        # First, precalculate the joined hosts so that the federation sender doesn't
+        # need to.
+        await self._event_creation_handler.cache_joined_hosts_for_event(event, context)
```

Comment:
Previously we would do this even if `context.rejected` was truth-y. I suspect that wasn't helpful since the event is rejected, but wanted to point this out as a change in behavior.

Also, I find the comment a bit odd "fIrst...", but there's no follow-up (no "then" / "next" / "second"). 😄

Reply:
> Previously we would do this even if context.rejected was truth-y. I suspect that wasn't helpful since the event is rejected, but wanted to point this out as a change in behavior.

yeah, it's deliberate, but I think the only difference is that we won't be populating a cache entry that will never be used.

> Also, I find the comment a bit odd "fIrst...", but there's no follow-up (no "then" / "next" / "second")

mmhmm. it's in reference to the line above: "we can persist the event ... But first, precalculate the joined hosts".

Reply:
Fair! Not a huge deal, just felt like I was left with a cliffhanger!